app/validators/blocks/calculated_summary_block_validator.py

```diff
@@ -1,59 +1,29 @@
-from app.validators.blocks.block_validator import BlockValidator
-from app.validators.questionnaire_schema import find_duplicates
+from app.validators.blocks.calculation_block_validator import CalculationBlockValidator
 
 
-class CalculatedSummaryBlockValidator(BlockValidator):
-    ANSWERS_MUST_HAVE_SAME_TYPE = (
-        "All answers in block's answers_to_calculate must be of the same type"
-    )
-    ANSWERS_MUST_HAVE_SAME_CURRENCY = (
-        "All answers in block's answers_to_calculate must be of the same currency"
-    )
-    ANSWERS_MUST_HAVE_SAME_UNIT = (
-        "All answers in block's answers_to_calculate must be of the same unit"
-    )
-    ANSWERS_HAS_INVALID_ID = "Invalid answer id in block's answers_to_calculate"
-    ANSWERS_HAS_DUPLICATES = "Duplicate answers in block's answers_to_calculate"
+class CalculatedSummaryBlockValidator(CalculationBlockValidator):
     ANSWER_SET_AFTER_CALCULATED_SUMMARY = (
         "Answer ids for calculated summary must be set before calculated summary block"
     )
     ANSWER_SET_IN_DIFFERENT_SECTION_FOR_CALCULATED_SUMMARY = "Answer ids for calculated summary must be set in the same section as the calculated summary block"
 
     def __init__(self, block, questionnaire_schema):
         super().__init__(block, questionnaire_schema)
-        self.answers_to_calculate = self.questionnaire_schema.get_calculated_answer_ids(
-            self.block
+        self.answers_to_calculate = self.questionnaire_schema.get_calculation_block_ids(
+            self.block, "answers"
```

Comment:
Its only two params but I'm enjoying the push to using kwargs when raw values are passed as args

Reply:
Nah happy with that suggestion, good idea 👍  
done